tests/lean/run/matchEqsBug.lean

```diff
@@ -26,12 +26,7 @@ set_option trace.Meta.Match.matchEqs true
 test% f.match_1
 #check f.match_1.splitter
 
-def g.match_1.splitter := 4
```

Comment:
The async code (must) assume that the same name isn't used by a private and a non-private declaration so this broke. I think that is an acceptable regression.